db/memtable.cc

```diff
@@ -434,7 +429,24 @@ FragmentedRangeTombstoneIterator* MemTable::NewRangeTombstoneIterator(
 
   auto* fragmented_iter = new FragmentedRangeTombstoneIterator(
       fragmented_tombstone_list, comparator_.comparator, read_seq);
-  return fragmented_iter;
+
+  fragmented_tombstone_iter = fragmented_iter;
+}
+
+FragmentedRangeTombstoneIterator* MemTable::NewRangeTombstoneIterator(
```

Comment:
It occurs to me now that this function should be renamed because calling code should not be trying to grab ownership over the returned pointer.  Perhaps also changing the return value to be a reference makes sense.

I also need to change any code that uses this function to reflect the above.